src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp

```diff
@@ -49,25 +53,44 @@ enum Yells
 enum Actions
 {
     ACTION_WATER_ELEMENT_HIT                    = 1,
-    ACTION_WATER_ELEMENT_KILLED                 = 2
 };
 
-/// @todo get those positions from spawn of creature 29326
-#define MAX_SPAWN_LOC 5
-static Position const SpawnLoc[MAX_SPAWN_LOC]=
+enum IchoronEvents
 {
-    {1840.64f, 795.407f, 44.079f, 1.676f},
-    {1886.24f, 757.733f, 47.750f, 5.201f},
-    {1877.91f, 845.915f, 43.417f, 3.560f},
-    {1918.97f, 850.645f, 47.225f, 4.136f},
-    {1935.50f, 796.224f, 52.492f, 4.224f}
+    EVENT_WATER_BLAST                           = 1,
+    EVENT_WATER_BOLT_VOLLEY,
+};
+
+enum GlobuleEvents
+{
+    EVENT_GLOBULE_MOVE                          = 1,
 };
 
 enum Misc
 {
+    DATA_GLOBULE_PATH                           = 0,
     DATA_DEHYDRATION                            = 1
 };
 
+Position globulePaths[10] =
```

Comment:
use a define to specify the size instead of a magic number and use the define in all the for loops